Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -64,6 +64,16 @@ extension BrowserViewController: WKNavigationDelegate {
         return false
     }
 
+    // This is the iOS 13 delegate callback which is similar to the delegate below it
+    // except that a WKWebpagePreferences is provided, Setting the type of contentMode loads the site accordingly.
+    @available(iOS 13.0, *)
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, preferences: WKWebpagePreferences, decisionHandler: @escaping (WKNavigationActionPolicy, WKWebpagePreferences) -> Void) {
+        self.webView(webView, decidePolicyFor: navigationAction) {
+            preferences.preferredContentMode = Preferences.General.alwaysRequestDesktopSite.value ? .desktop :. mobile
+            decisionHandler($0, preferences)
```

Comment:
This breaks `requestMobileSite` from the Share Menu because it only obeys your "settings" from `Preferences` and not your site's preference..

IE: Have AlwaysRequestDesktop site on.. Then go to youtube and go to share menu and choose `RequestMobileSite`, it will still show Desktop, but if you go back to that same share menu, it will show `RequestDesktopSite` even though you're already on desktop UA..

In other words, the text will change for the share menu button but it does nothing in the end.

Reply:
Its not because of the settings but somehow YouTube is setting the url param as app==desktop. If you instead try  https://www.whatsmyua.info or anyother site it shows it works fine.

Reply:
let me see if I can fix it on our end

Reply:
I found that the reason is because we are trying to open a mobile site ie `m.youtube.com` in Desktop. Youtube then redirects it to `https://www.youtube.com/?app=desktop`. When we change the UA with the param app=desktop no change is done. 

I propose that we also change our favourites URL from mobile to desktop in this case? @Brandon-T  @iccub @jhreis

Reply:
It happens on Desktop as well

Reply:
It doesn't work for me.
I open https://www.whatsmyua.info, request mobile site, and doesn't switch to mobile

I agree we can remove `mobile` part from favorites, youtube and twitter